File: datacollector/src/main/java/de/leo/fingerprint/datacollector/utils/TimeUtils.java
package de.leo.fingerprint.datacollector.utils;

import org.threeten.bp.LocalDateTime;
import org.threeten.bp.format.DateTimeFormatter;

/**
 * Created by Leo on 4/24/2016.
 */
public class TimeUtils {

    public static String getCurrentTimeStr() {
        return LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"));
    }

    public static String getTimeStr(LocalDateTime date) {
        return date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"));
    }

    public static LocalDateTime getDateFromString(String timeStr) {
        return LocalDateTime.parse(timeStr, DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"));
    }
}

File: datacollector/src/main/java/de/leo/fingerprint/datacollector/utils/TimeUtils.kt
package de.leo.fingerprint.datacollector.utils

import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZonedDateTime
import org.threeten.bp.format.DateTimeFormatter
import java.util.concurrent.TimeUnit

/**
 * Created by Leo on 4/24/2016.
 */
object TimeUtils {

    val currentTimeStr: String
        get() = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"))

    fun getTimeStr(date: LocalDateTime): String {
        return date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"))
    }

    fun getDateFromString(timeStr: String): LocalDateTime {
        return LocalDateTime.parse(timeStr, DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"))
    }

    fun ZonedDateTime.toEpochMillis(): Long = TimeUnit.SECONDS.toMillis(toEpochSecond())
}
